docs/contribute/colima.qmd

```diff
@@ -0,0 +1,130 @@
+you can run docker in Mac m1/m2 without Docker Desktop.
+
+## Colima
+
+[Colima](https://github.com/abiosoft/colima) is container runtimes on macOS
+(and Linux) with minimal setup, that has M1 macs support.
+
+### Get started
+
+1. Uninstall Docker Desktop
+
+If you have Docker desktop installed, followed the references, both of them have
+a section on how to get rid of it.
+
+2. Install Docker Client with `brew`
+
+Make sure you brew is up to date and cashed cleaned up, just in case.
+
+```bash
+$ brew install docker
+```
+Check installation by doing
+
+```bash
+$ docker --version
+```
+
+```bash
+$ docker version
+```
+
+Notice we haven't installed any Docker Engine yet.
+
+3. Install Colima
+
+```bash
+$ brew install colima
+```
+
+4. Install Docker Compose Plugin (optional)
+
+If you want to be able to run `docker compose ...` follow this steps. Noticed I
+said `docker compose ...` not `docker-compose ...`
+
+According to the official Docker [documentation](https://docs.docker.com/compose/install/linux/),
+run the following commands in a separate terminal:
+
+Replace the link on the `curl` step with the version you desired, find them here
+https://github.com/docker/compose/releases/
+
+```bash
+$ DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
+$ mkdir -p $DOCKER_CONFIG/cli-plugins
+$ curl -SL https://github.com/docker/compose/releases/download/v2.24.6/docker-compose-darwin-aarch64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
+```
+
+Then add execution permission to the downloaded binary:
+
+```bash
+$ chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
+```
+
+Check it worked:
+
+```bash
+$ docker compose version
+```
+
+5. Take it for a spin. (Example for the [Ibis](https://github.com/ibis-project/ibis) project)
+
+```bash
+$ colima start
+```
+Note:
+By default this starts with 2cpu and 2GB of memory, and 60GB disk
+
+```bash
+$ colima status
+$ colima list
+```
+
+You can modify this, with `--cpu`, `--memory` and `--disk`.
+
+In the Ibis repo base directory
+
+```bash
+$ just up postgres
+```
+
+This will trigger a docker compose up. Once the container is up and reports a
+healthy status, you can run the tests by doing:
+
+bash
+```
+$ pytest -m postgres
+```
+Once you are done, you can stop the container by doing:
+
+```bash
+$ just down postgres
+```
+If you are done for the day, and want to avoid the Colima instance eating your
+resources, you will want to stop it.
+
+```bash
+$ colima stop
+```
+
+You can start colima again with just `colima start`. If you upgraded colima,
+you need to delete the existing instance by doing. I also found that if you want
+to modify the allocation, you need to delete the existing instance too (unless
+you are going only up).
+
+```bash
+$ colima delete
+```
+
+**Extra notes**
+
+- You can provide `--arch` to `colima start` to specify an architecture.
+- You can modify the `--runtime` (by default `docker`) to be `conteinerd`.
+- You can enable kubernetes
+
+**References:**
```

Comment:
Quarto has [citations and footnotes](https://quarto.org/docs/authoring/footnotes-and-citations.html) you can use for this, and they'll be rendered will backwards and forwards references.